src-gui/src/renderer/components/modal/seed-selection/SeedSelectionDialog.tsx

```diff
@@ -107,14 +114,71 @@ export default function SeedSelectionDialog() {
     return null;
   }
 
-  // Disable the button if the user is restoring from a seed and the seed is invalid
-  // or if selecting wallet path and no path is selected
+  // Disable the button if the user is restoring from a seed and the seed is invalid,
+  // if selecting wallet path and no path is selected,
+  // or if setting a password and they don't match
   const isDisabled =
     selectedOption === "FromSeed"
       ? customSeed.trim().length === 0 || !isSeedValid
       : selectedOption === "FromWalletPath"
         ? !walletPath
-        : false;
+        : selectedOption === "RandomSeed" && password !== passwordRepeat;
+
+  const handleTogglePasswordVisibility = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const passwordInputs = (
```

Comment:
this be its own component, this way we can also move the other state (setShowPassword, passwordRepeat, passwordError, ...) into that component